### DISCUSSION TITLE
Close all open panes that are not pinned

### DISCUSSION THREAD

**Discussion by welpdx** [SilentVoid13/Templater#682] [Templates Showcase]
If you want a quick way to close all the opened panes (except the ones you've pinned), you can use this templater script:
### Script
```js
<%* 
var list = []
this.app.workspace.iterateRootLeaves((s) => {
	list.push(s);
})
for (var i = 0; i < list.length; i++) {
    if (list[i].getViewState()?.pinned != true) {
        list[i].detach()
    }
} 
%>
```

### demo: 
![img](https://i.imgur.com/r0zLOHM.gif)

### potential workflow
Save all opened panes using Obisidian Workspaces
Then, close all unpinned panes to work on a diff task in your vault.

**Comment by Nusnus** [comment]
Cool one! Thanks!
